onnxruntime/core/framework/tunable.h

```diff
@@ -164,6 +176,19 @@ class TunableOp {
     default_id_ = id;
   }
 
+  void RegisterTunableSubOp(TunableOp<ParamsT, TimerT>* op_ptr) {
```

Comment:
How about `void RegisterTunableOp(TunableOp<ParamsT, TimerT>&&)`, notice the name change and transfer the onwership. Then deregister can also be deleted

In the future, we may add `RegisterOp` to replace our current ad hoc `this->ops_.emplace_back` calls...

Reply:
Thanks for the suggestion. I've Renamed it as `nested_tunable_ops_`, and the method names as `RegisterNestedTunableOp` and `DeregisterNestedTunableOp`. As for transferring the ownership: how would I actually _use_ the nested tunable op in the subclass if I've transferred ownership to the base?